include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -339,12 +344,28 @@ class fm_index
                           | views::join(delimiter),
                           std::ranges::begin(tmp_text));
 
+        if (!reverse)
+        {
+            // we need at least one delimiter
+            if (number_of_texts == 1)
+                tmp_text.back() = delimiter;
 
-        // we need at least one delimiter
-        if (std::ranges::distance(text) == 1)
-            tmp_text.back() = delimiter;
-
-        std::ranges::reverse(tmp_text);
+            std::ranges::reverse(tmp_text);
```

Comment:
why are we reversing the text here in the `if (!reverse)` branch?

Reply:
the `reverse` is whether the original text was reversed (reverse_fm_index).
If it's a "normal" fm_index, you can just reverse the test; if the text comes from the `reverse_fm_index` you have to do special stuff if there is only one item in the collection